scheduler_test.go

```diff
@@ -361,6 +361,42 @@ func TestScheduler_StopTimeout(t *testing.T) {
 	}
 }
 
+func TestScheduler_StopLongRunningJobs(t *testing.T) {
+	t.Run("start, run job, stop jobs before job is completed", func(t *testing.T) {
+		s := newTestScheduler(t,
+			WithStopTimeout(50*time.Millisecond),
+		)
+
+		_, err := s.NewJob(
+			DurationJob(
+				50*time.Millisecond,
+			),
+			NewTask(
+				func(ctx context.Context) {
+					select {
+					case <-ctx.Done():
+						time.Sleep(time.Second)
```

Comment:
This test is failing, which makes sense since on cancellation the job is sleeping here, which would prevent it from stopping.

Is there an intention I'm missing here?
```suggestion
```